crates/sui/src/upgrade_compatibility.rs

```diff
@@ -499,12 +502,13 @@ pub(crate) async fn check_compatibility(
     let policy =
         UpgradePolicy::try_from(upgrade_policy).map_err(|_| anyhow!("Invalid upgrade policy"))?;
 
-    compare_packages(existing_modules, new_package, policy)
+    compare_packages(existing_modules, new_package, package_path, policy)
 }
 
 fn compare_packages(
     existing_modules: Vec<CompiledModule>,
     new_package: CompiledPackage,
```

Comment:
We already had to parse the files for `CompiledPackage`. 
It would probably be best as a separate PR, but we should just save the file map in there so we don't have to reread the files in here.

Reply:
done